### PR TITLE
Set configdata in M2 from database

### LINKED ISSUE #227: [TASK] Autconfig WSC - fill in M2 information from database
## Description

The Intel vendor-specific architecture doesn't really do autoconfiguration. Currently, the M2 of the autoconfiguration is just filled in with fixed data. For certification, it needs to be filled in with real data.

## Exit criteria

- [ ] Controller database contains a list of ALIDs, operating classes, SSIDs and credentials to configure.
- [ ] This table is initialised with the static data currently used (for testing).
- [ ] Autoconfig M1 handling matches the band to the table in the database and constructs M2s based on that.
- [ ] Agent reports the details of the received M2 (for testing).
- [ ] DEV_RESET_DEFAULT resets the credentials in the database.
- [ ] DEV_SET_CONFIG adds credentials to the database.
- [ ] test_flows.sh includes a test for both of the above.

### PR DESCRIPTION
**Pull request by arnout**
Closes #227

There is no test yet. The easiest way to test this is by adding support for AUTOCONFIG_RENEW to the agent, so that we can re-trigger autoconfiguration in one of the test flows. But that's a bigger change so I'll do it in a separate PR.

### CODE REVIEW

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [inline controller/src/beerocks/master/son_master_thread.cpp]: This will have to be changed after #251 will be merged.

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [inline controller/src/beerocks/master/son_master_thread.cpp]: What is "@a" ?

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [inline controller/src/beerocks/master/son_master_thread.cpp]: In the past, there was a decision no to add static/helper functions to the son_master_thread since it was big enough, and instead, put them on son_actions.
I don't say it is right, but it is a convention.

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [inline controller/src/beerocks/master/son_master_thread.cpp]: I see the spell of BSS in a plural term is "bsss" (funny IMO), on other places I see BSSes.

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [inline controller/src/beerocks/master/son_master_thread.cpp]: I assumed you meant not to reach here if you called `autoconfig_wsc_add_m2(tlvwscM1, &bss_info_conf)` in line 742, but we will reach here after the previous ranges loop will be finished.

**Review by morantr — DISMISSED** [review]

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (tomereli) [reply controller/src/beerocks/master/son_master_thread.cpp]: So this should be wrapped with
```
if (!num_bsses)
   ....
```

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (tomereli) [inline controller/src/beerocks/master/son_master_thread.cpp]: We should refer to v2.0.6 which is the WSC specification referred to by the EasyMesh Specification, see section 2 
>[5]	Wi-Fi Simple Configuration Technical Specification 2.0.6 (https://www.wi-fi.org/members/certification-programs )

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (morantr) [reply controller/src/beerocks/master/son_master_thread.cpp]: @tomereli, try to read the brief in line 379, you will understand what I meant.

Comment (arnout) [reply controller/src/beerocks/master/son_master_thread.cpp]: @a is a command to format the following word as code. It is meant to be used for arguments to functions, since those are not auto-formatted (unlike class and function references).

Cfr. http://doxygen.nl/manual/commands.html#cmda

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (arnout) [reply controller/src/beerocks/master/son_master_thread.cpp]: > This will have to be changed after #251 will be merged.

That's why it's still a draft :-)

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (arnout) [reply controller/src/beerocks/master/son_master_thread.cpp]: > I see the spell of BSS in a plural term is "bsss" (funny IMO), on other places I see BSSes.

Indeed, `maximum_number_of_bsss_supported` already exists.

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (arnout) [reply controller/src/beerocks/master/son_master_thread.cpp]: Oops....

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (arnout) [reply controller/src/beerocks/master/son_master_thread.cpp]: Meh, I don't have that version of the document and I can't access the WFA site. Can you confirm that the references are still correct?

And then I should try and get version 2.0.6...

**Comment by arnout** [comment]
I will not be able to continue working on this tomorrow, so if anyone wants to take this over on Sunday, feel free.

One thing that is still missing, is updating test_flows.sh with something to actually test this. My idea was to follow the current tests with DEV_RESET_DEFAULT and a sample DEV_SET_CONFIG taken from Table 11, then restart the repeater, then grep on the repeater if it receives the M2 with the correct SSID.

File: framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml
Comment (morantr) [inline framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml]: _value_const: TEARDOWN

Comment (arnout) [reply framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml]: Apparently, _value_const is not supported in struct...

File: controller/src/beerocks/master/son_master_thread.cpp
Comment (tomereli) [reply controller/src/beerocks/master/son_master_thread.cpp]: The references are still correct.

**Review by tomereli — APPROVED** [review]

**Comment by tomereli** [comment]
@arnout - @morantr is OOO today so if you are ready to push this, you can dismiss his review I assume.

**Pull request merged**
